src/plugins/intel_gna/samples/transformer/src/app/menu.cpp

```diff
@@ -0,0 +1,173 @@
+    init(argc, argv);
+}
+
+static void print_usage() {
+    log_info() << std::endl;
+    log_info() << "gna_transformer_app [OPTION]" << std::endl;
+    log_info() << "Options:" << std::endl;
+    log_info() << std::endl;
+    log_info() << "    -h                            " << help_message << std::endl;
+    log_info() << "    -input_model \"<path>\"       " << input_model_message << std::endl;
+    log_info() << "    -output_model \"<path>\"      " << output_model_message << std::endl;
+    log_info() << "    -config \"<path>\"            " << config_message << std::endl;
+    log_info() << "    -transformation \"<string>\"  " << transformation_message << std::endl;
+    log_info() << "    -list_app_transformations     " << list_app_transformations_message << std::endl;
+    log_info() << std::endl;
+    log_info() << "    commands examples:" << transformation_message << std::endl;
+    log_info() << std::endl;
+
+    log_info() << example_command_config << std::endl;
+    log_info() << example_command_transformation << std::endl;
+}
+
+void Menu::execute_action() {
+    if (m_action) {
+        m_action();
+        log_info() << "Transformation successed" << std::endl;
+    }
+}
+
+void Menu::init(int argc, char* argv[]) {
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        print_usage();
+        return;
+    }
+
+    if (FLAGS_list_app_transformations) {
+        print_app_defined_transformations();
+        return;
+    }
+
+    if (FLAGS_input_model.empty()) {
+        throw WrongArgumentsExeception("Lack of input_model argument. Please check -h option");
+    }
+
+    if (FLAGS_output_model.empty()) {
+        throw WrongArgumentsExeception("Lack of output_model argument. Please check -h option");
+    }
+
+    if (!FLAGS_config.empty()) {
+        m_action = prepare_gna_specific_transformations(FLAGS_config, FLAGS_input_model, FLAGS_output_model);
+        return;
+    }
+
+    if (!FLAGS_transformation.empty()) {
+        m_action = prepare_app_specific_transformation(FLAGS_transformation, FLAGS_input_model, FLAGS_output_model);
+        return;
+    }
+
+    throw WrongArgumentsExeception("Lack of one of arguments: config or transformation. Please check -h option");
+}
+
+std::function<void(void)> Menu::prepare_app_specific_transformation(const std::string& transformation_name,
+                                                                    const std::string& input_model_path,
+                                                                    const std::string& output_model_path) {
+    log_info() << "Chosen Application defined transformation: " << transformation_name << "." << std::endl;
+
+    passes::AppPassManager manager;
+
+    auto pass = manager.get_pass(transformation_name);
+
+    std::shared_ptr<ModelTransformer> transformer = std::make_shared<ModelTransformerApp>(std::move(pass));
+
+    return [this, input_model_path, transformer, output_model_path]() {
```

Comment:
```suggestion
    return [input_model_path, transformer, output_model_path]() {
```

Reply:
warranty for Menu's longer life time are no longer needed (converting other two methods above to static are involved in here)